pkg/i2gw/providers/common/converter.go

```diff
@@ -319,7 +319,23 @@ func (rg *ingressRuleGroup) configureBackendRef(paths []ingressPath) ([]gatewayv
 	var errors field.ErrorList
 	var backendRefs []gatewayv1beta1.HTTPBackendRef
 
+	// track unique backends to avoid duplicates
+	uniqueBackends := make(map[string]interface{})
```

Comment:
Since the intended functionality here is like a set, I suggest using map[string]struct{} to reduce memory usage (0 vs 16 bytes per entry).